lib/azure/AzureDataStore.ts

```diff
@@ -59,39 +47,37 @@ export default class AzureDataStore extends iDatastore.Adapter {
   }
   /**
    * Returns the full key which includes the path to the ipfs store
-   * @param key
+   * @param key The key to get the full path for
    */
   private getFullKey (key: any): string {
     return path.join('.', this.path, key.toString());
   }
 
   /**
-   * Recursively fetches all keys from azure blob storage
-   * @param params
-   * @param keys
-   * @param callback
+   * Fetches keys from storage
+   * @param prefix The prefix to filter for
+   * @param currentToken The current token to know where in the index to start
+   * @return an object containing continuation token indicating there are more keys to retrieve, and a list of keys
    */
-  private async listKeys (prefix: string, currentToken: any, keys: any): Promise<any> {
+  private async listKeys (prefix: string, currentToken: any): Promise<any> {
     return new Promise((resolve, reject) => {
       this.blobService.listBlobsSegmentedWithPrefix(this.container, prefix, currentToken, async (err, result, response) => {
         if (err) {
           reject(new Error(err.name));
           return;
         }
         if (response.isSuccessful) {
+          const keys = [];
           result.entries.forEach((d) => {
```

Comment:
Sorry forgot to comment explicitly earlier: this needs to be a synchronous for loop.

Reply:
My bad, foreach() is sync.